contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -267,64 +267,65 @@ contract LinkAvailableBalanceMonitor is AccessControl, AutomationCompatibleInter
       address targetAddress = targetAddresses[idx];
       contractToFund = s_targets[targetAddress];
       s_targets[targetAddress].lastTopUpTimestamp = uint56(block.timestamp);
-      if (
-        localBalance >= contractToFund.topUpAmount &&
-        _needsFunding(
-          targetAddress,
-          contractToFund.lastTopUpTimestamp + minWaitPeriod,
-          contractToFund.minBalance,
-          contractToFund.isActive
-        )
-      ) {
-        bool success = i_linkToken.transfer(targetAddress, contractToFund.topUpAmount);
+
+      (bool fundingNeeded, address target) = _needsFunding(
+        targetAddress,
+        contractToFund.lastTopUpTimestamp + minWaitPeriod,
+        contractToFund.minBalance,
+        contractToFund.isActive
+      );
+      if (localBalance >= contractToFund.topUpAmount && fundingNeeded) {
+        bool success = i_linkToken.transfer(target, contractToFund.topUpAmount);
         if (success) {
           localBalance -= contractToFund.topUpAmount;
-          emit TopUpSucceeded(targetAddress);
+          emit TopUpSucceeded(target, contractToFund.topUpAmount);
         } else {
-          s_targets[targetAddress].lastTopUpTimestamp = contractToFund.lastTopUpTimestamp;
+          s_targets[targetAddress].lastTopUpTimestamp = contractToFund.lastTopUpTimestamp; // should we update this?
```

Comment:
@RyanRHall  what do you think of this?

Reply:
this is super confusing and unnecessarily complicated lol. From what I can tell, the `lastTopUpTimestamp` is _always_ updated on line 269, and then on line 283 and 287 we set the time stamp back to what it was (meaning net no change) - is that how you interpret this code too?

I think we should take a second and think about what makes most sense from a product perspective. The min wait period is supposed to act as a rate limiter. And I think the question we're asking here is "should the rate limit apply to failed transfers?". So imagine a transfer fails for some reason, do we want the system to retry immediately or only retry after some wait period? I would think the latter, but we might want to double check with finance team.